servicetalk-benchmarks/src/jmh/java/io/servicetalk/data/jackson/JacksonStreamingSerializerBenchmark.java

```diff
@@ -0,0 +1,177 @@
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static io.servicetalk.concurrent.api.Publisher.fromIterable;
+import static io.servicetalk.data.jackson.JacksonSerializerFactory.JACKSON;
+
+/**
+ * Performs (de)serialization benchmarks on the {@link JacksonStreamingSerializer}.
+ * <p>
+ * Benchmark                                                            Mode  Cnt        Score      Error  Units
+ * JacksonStreamingSerializerBenchmark.deserializeLargeBackedByArray   thrpt    5   102204,872 ±  391,655  ops/s
+ * JacksonStreamingSerializerBenchmark.deserializeLargeBackedByDirect  thrpt    5   102117,249 ±  396,363  ops/s
+ * JacksonStreamingSerializerBenchmark.deserializeMidBackedByArray     thrpt    5   232416,219 ±  597,317  ops/s
+ * JacksonStreamingSerializerBenchmark.deserializeMidBackedByDirect    thrpt    5   277589,410 ±  574,916  ops/s
+ * JacksonStreamingSerializerBenchmark.deserializeSmallBackedByArray   thrpt    5  1123782,216 ± 3798,021  ops/s
+ * JacksonStreamingSerializerBenchmark.deserializeSmallBackedByDirect  thrpt    5  1137170,185 ± 5476,999  ops/s
+ * <p>
+ * The numbers above show streaming deserialization of different payload sizes, divided into a total of 8 chunks per
+ * stream decode. Small is 10 bytes, Mid is 1KB and Large is 1MB of a single JSON object with a random number of
+ * fields and values inside.
+ */
+@Fork(value = 1)
+@State(Scope.Benchmark)
+@Warmup(iterations = 5, time = 3)
+@Measurement(iterations = 5, time = 3)
+@BenchmarkMode(Mode.Throughput)
+public class JacksonStreamingSerializerBenchmark {
+
+    /**
+     * Allows to customize the number of chunks the payload is split up to simulate JSON chunk stream decoding.
+     */
+    private static final int NUM_CHUNKS = 8;
+
+    /**
+     * Type reference used for deserialization mapping.
+     */
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE =
+            new TypeReference<Map<String, Object>>() {
+            };
+
+    private static final Buffer SMALL_JSON_HEAP = BufferAllocators.PREFER_HEAP_ALLOCATOR.fromUtf8(generateJson(10));
+    private static final Buffer SMALL_JSON_DIRECT =
+            BufferAllocators.PREFER_DIRECT_ALLOCATOR.wrap(SMALL_JSON_HEAP.toNioBuffer());
+
+    private static final Buffer MID_JSON_HEAP = BufferAllocators.PREFER_HEAP_ALLOCATOR.fromUtf8(generateJson(1_000));
+    private static final Buffer MID_JSON_DIRECT =
+            BufferAllocators.PREFER_DIRECT_ALLOCATOR.wrap(MID_JSON_HEAP.toNioBuffer());
+
+    private static final Buffer LARGE_JSON_HEAP =
+            BufferAllocators.PREFER_HEAP_ALLOCATOR.fromUtf8(generateJson(1_000_000));
+    private static final Buffer LARGE_JSON_DIRECT =
+            BufferAllocators.PREFER_DIRECT_ALLOCATOR.wrap(LARGE_JSON_HEAP.toNioBuffer());
```

Comment:
The number are not different much between heap/direct because you always use heap buffers :) 
Have a look inside `wrap` method.
In order to get a direct memory, you need to use either `PREFER_DIRECT_ALLOCATOR.fromUtf8` method or  `PREFER_DIRECT_ALLOCATOR.newBuffer(...)` and then copy content from `LARGE_JSON_HEAP`.
When you adjust, consider using the same content (cache `generateJson` result or copy from heap to direct).